### PR TITLE
Interactivity and Dynamic UIs Docs page updated with ES6 Example

### PR DESCRIPTION
**Pull request by vedatmahir**
ES5 example on this page updated with ES6.

### CODE REVIEW

File: docs/docs/03-interactivity-and-dynamic-uis.md [hunk 1]
```diff
@@ -11,22 +11,25 @@ You've already [learned how to display data](/react/docs/displaying-data.html) w
...
-  },
-  handleClick: function(event) {
+class LikeButton extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      liked: false
+    }
+  }
+  handleClick = () => {
```
Comment (jimfb) [inline docs/docs/03-interactivity-and-dynamic-uis.md]: We won’t use class properties in the docs. We only want to use stable ES2015 syntax. So if we need to bind an event handler, we will do this in the constructor.

Comment (jimfb) [inline docs/docs/03-interactivity-and-dynamic-uis.md]: You can just make this a standard function, and in the constructor say `this.handleClick = this.handleClick.bind(this);`

**Comment by facebook-github-bot-7** [comment]
@vedatmahir updated the pull request.

**Comment by jimfb** [comment]
Thanks!

File: docs/docs/03-interactivity-and-dynamic-uis.md [hunk 2]
```diff
@@ -11,22 +11,26 @@ You've already [learned how to display data](/react/docs/displaying-data.html) w
...
 
 ```javascript
-var LikeButton = React.createClass({
-  getInitialState: function() {
-    return {liked: false};
-  },
-  handleClick: function(event) {
+class LikeButton extends React.Component {
+  constructor() {
+    super();
```
Comment (gaearon) [inline docs/docs/03-interactivity-and-dynamic-uis.md]: Shall we pass props always for consistency?

Comment (jimfb) [inline docs/docs/03-interactivity-and-dynamic-uis.md]: Yeah, I thought about that.  But if we're always going to pass props, why not context?

Also, what's the benefit of context, when 95% of the time, you don't need to access props.

I convinced myself that it was better to only pass them/use when you need them.

**Pull request merged**